app/models/pieces/king.rb

```diff
@@ -1,4 +1,23 @@
 class King < Piece
+  def can_castle?(rook)
+    return false unless first_move == 'true'
+
+    return false unless rook.first_move == 'true'
+
+    return false if is_obstructed?(rook.x, rook.y)
+
+    true
+  end
+
+  def castle!(rook)
+    x_coords = current_to_destination_x_coordinates(rook.x)
+
+    if x_coords.size == 3 || x_coords.size == 2
+      update(x: x_coords[1])
+      rook.update(x: x_coords[0])
+    end
+  end
```

Comment:
you still have the `castle` stuff in here.

Probably happened because you branched off of your castling branch.

Each time you start a new branch, I recommend going back to master

```
git checkout master
git pull
git checkout -b new-branch
```